Polyhedron/demo/Polyhedron/Plugins/Operations_on_polyhedra/Clip_polyhedron_plugin.cpp

```diff
@@ -173,58 +175,112 @@ public Q_SLOTS:
     {
       QApplication::setOverrideCursor(Qt::WaitCursor);
       QGLViewer* viewer = *QGLViewer::QGLViewerPool().begin();
-      QList<Scene_polyhedron_item*> polyhedra;
+      QList<Scene_item*> polyhedra;
 
       //Fills the list of target polyhedra and the cutting plane
       Q_FOREACH(int id, scene->selectionIndices())
       {
-        Scene_polyhedron_item *target_item = qobject_cast<Scene_polyhedron_item*>(scene->item(id));
-        if(target_item)
+        Scene_surface_mesh_item *sm_item = qobject_cast<Scene_surface_mesh_item*>(scene->item(id));
+        if(sm_item && CGAL::is_triangle_mesh(*sm_item->polyhedron()))
+        {
+          polyhedra << sm_item;
+        }
+        else
         {
-          polyhedra << target_item;
+          Scene_polyhedron_item *poly_item = qobject_cast<Scene_polyhedron_item*>(scene->item(id));
+          if(poly_item && CGAL::is_triangle_mesh(*poly_item->polyhedron()))
+          {
+            polyhedra << poly_item;
+          }
         }
       }
-
       //apply the clipping function
-      Q_FOREACH(Scene_polyhedron_item* poly, polyhedra)
+      Q_FOREACH(Scene_item* item, polyhedra)
       {
+        Scene_surface_mesh_item *sm_item = qobject_cast<Scene_surface_mesh_item*>(item);
+        Scene_polyhedron_item* poly_item = NULL;
+        if(!sm_item)
+        {
+          poly_item = qobject_cast<Scene_polyhedron_item*>(item);
+        }
```

Comment:
This could be simplified to:
```C++
Scene_surface_mesh_item* sm_item = qobject_cast<Scene_surface_mesh_item*>(item);
Scene_polyhedron_item* poly_item = qobject_cast<Scene_polyhedron_item*>(item);
```
One of `sm_item`/`poly_item` will be null, and the other will be non-null.